File: problem_3.py
# What is the largest prime factor of the number 600851475143 ? 
# 1/6/13 

import math

def populate(list, n):
	start1 = 0
	cap1 = (int)(n/2)
	start2 = cap1
	cap2 = (int)(n)
	print "cap1:", cap1
	print "start2:", start2

	for i in xrange (start1, cap1):
		n[i] = True
	for j in xrange (start2, cap2):
		n[j] = True

limit = 600851475143
nums = [] # BOOLEAN array!!!!!
nums = populate(nums, limit)
  
pos = 0 # corresponds to n = 2 
while pos < (int)(math.ceil(math.sqrt(limit))): 
    if nums[pos] == False: 
        pos += 1
    else: 
        for i in range (2*pos + 2, len(nums), pos + 2): 
            nums[i] = False
        pos += 1

prime_factors = []
prime_num = 0
for i in range (len(nums)): 
    if nums[i] == True: 
        prime_num = i + 2
        if limit % prime_num == 0:
        	prime_factors.append(prime_num) 

print max(prime_factors)

''' 
total = 0 
number = 0 
for i in range (len(nums)): 
    if nums[i] == True: 
        number = i + 2 
        total += number 
'''

File: Problem3.java
/* WE MUST GO WITH THE JAVA VERSION FOR PROBLEM 3!
 * 1/6/13
 */

/*public class Problem3{
	public static Boolean[] populate(Boolean[] list, long n){
		long cap1 = (long)(Math.ceil(n));
		long cap2 = n;
		int i = 0;

		while (i <= cap1){
			list[i] = true;
			i++;
		}

		System.out.println("i is now" + i);
		return null;
	}


	public static void main(String[] args){
		long limit = 600851475143L;
		System.out.println(Long.MAX_VALUE);
		//Boolean[] nums = new Boolean[limit];
		//populate(nums, limit);
		//System.out.println(nums.size());
	}
}*/
import java.util.*;
public class Problem3{
	public static Boolean[] populate(ArrayList<Boolean> list, long n){
		//long cap1 = (long)(Math.ceil(n));
		//long cap2 = n;
		int i = 0;
		while (i <= n){
			list.add(true);
			i++;
		}

		System.out.println("Length of arraylist: " + list.size());
		return null;
	}


	public static void main(String[] args){
		long limit = 100;
		//System.out.println(Long.MAX_VALUE);
		ArrayList<Boolean> nums = new ArrayList(); // should grow as we go
		populate(nums, limit);
		//System.out.println(nums.size());
	}
}